### PR TITLE
Switch to readdir() from readdir_r()

### PR DESCRIPTION
**Pull request by linkmauve**
Old form is deprecated, this fixes a warning on gcc.

### CODE REVIEW

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-09-11T04:04:04Z: https://github.com/citra-emu/citra/compare/722af0703efd58d2c2de9edbac763355053601a2...linkmauve:928c8a5206da72410fbb343d9e676fc99d4f724b

File: src/common/file_util.cpp [hunk 1]
```diff
@@ -457,14 +457,13 @@ bool ForeachDirectoryEntry(unsigned* num_entries_out, const std::string &directo
...
 #else
-    struct dirent dirent, *result = nullptr;
-
     DIR *dirp = opendir(directory.c_str());
     if (!dirp)
         return false;
 
     // non windows loop
-    while (!readdir_r(dirp, &dirent, &result) && result) {
+    struct dirent *result;
```
Comment (wwylele) [inline src/common/file_util.cpp]: `direct* result;`

File: src/common/file_util.cpp [hunk 2]
```diff
@@ -560,12 +559,11 @@ void CopyDir(const std::string &source_path, const std::string &dest_path)
     if (!FileUtil::Exists(source_path)) return;
     if (!FileUtil::Exists(dest_path)) FileUtil::CreateFullPath(dest_path);
 
-    struct dirent dirent, *result = nullptr;
     DIR *dirp = opendir(source_path.c_str());
     if (!dirp) return;
 
-    while (!readdir_r(dirp, &dirent, &result) && result)
-    {
+    struct dirent *result;
```
Comment (wwylele) [inline src/common/file_util.cpp]: ditto

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-09-11T04:16:54Z: https://github.com/citra-emu/citra/compare/722af0703efd58d2c2de9edbac763355053601a2...linkmauve:6c35e78dbba81f8839b74347a732bd847117ed67

File: src/common/file_util.cpp [hunk 3]
```diff
@@ -457,14 +457,13 @@ bool ForeachDirectoryEntry(unsigned* num_entries_out, const std::string &directo
...
-    struct dirent dirent, *result = nullptr;
-
     DIR *dirp = opendir(directory.c_str());
     if (!dirp)
         return false;
 
     // non windows loop
-    while (!readdir_r(dirp, &dirent, &result) && result) {
+    struct dirent* result;
+    while ((result = readdir(dirp))) {
```
Comment (yuriks) [inline src/common/file_util.cpp]: Double parenthesis

File: src/common/file_util.cpp [hunk 4]
```diff
@@ -560,12 +559,11 @@ void CopyDir(const std::string &source_path, const std::string &dest_path)
...
     if (!FileUtil::Exists(dest_path)) FileUtil::CreateFullPath(dest_path);
 
-    struct dirent dirent, *result = nullptr;
     DIR *dirp = opendir(source_path.c_str());
     if (!dirp) return;
 
-    while (!readdir_r(dirp, &dirent, &result) && result)
-    {
+    struct dirent* result;
+    while ((result = readdir(dirp))) {
```
Comment (yuriks) [inline src/common/file_util.cpp]: Double parenthesis

File: src/common/file_util.cpp (hunk 3 above)
Comment (yuriks) [inline src/common/file_util.cpp]: Also, you can declare the variable inside the loop: `while (struct dirent* result = readdir(dirp)) {`

**Comment by yuriks** [comment]
Othe than the code nits, LGTM. Gotta love the wonderful world of POSIX APIs...

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-09-13T13:11:35Z: https://github.com/citra-emu/citra/compare/f4bb89ce7397ec51c4af64fa70ec439befd4d38f...linkmauve:519e7013be9519113117221e6f00404d8685be51

**Comment by linkmauve** [comment]
Fixed those two.

Btw, double parentheses are a common idiom to silence the “hey, are you sure you didn’t actually mean to do an equal comparison here?” warning.

**Comment by yuriks** [comment]
Ah, wasn't aware of that, good to know.

**Pull request merged**